app/js/enhancedDate.js

```diff
@@ -1,5 +1,113 @@
 /* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+ * This module will return an object with each of these methods defined on it. 
+ * This module should have a notion of "private" data, that is data that is used only internally by the module and is not exposed aspart of the public API.
+ * The module should handle the case when an API method (other than setDate) is called before a date has been set. 
+ * One approach to this is to check if a date has already been set, and if not, set a new date at that time. 
+ * Only the setDate method should be able to change the internally stored date after it has been set once.
+*/
+
+var enhancedDate = (function enhancedDate() {
+
+    'use strict';
+
+    //Private data, which is data that is used only internally by the module and is not exposed as part of the public API
+    var privateData = {
+        days : ['Sunday', 'Monday', 'Tuesday', 'Wednesday',
+                     'Thursday', 'Friday', 'Saturday'],
+        months : ['January', 'February', 'March', 'April', 'May',
+                   'June', 'July', 'August', 'September', 'October',
+                   'November', 'December']
+    };
+
+    // check if a date has already been set, and if not, set a new date at that time.
+    function checkHasDate(){
+    
+        if (!privateData.storedDate) {
+            privateData.storedDate = new Date();
+        }
+    }
+
+    var publicAPI = {
+
+        /*
+         * Stores the passed-in date. This function can take milliseconds after epoch or a Date object.  
+         * Do some type-checking to make sure you have a valid type. If no value is passed, default your internally stored date to now.
+        */
+        setDate: function setDate(inputDate) {
+
+            checkHasDate();
+
+            if (inputDate){
+
+                // check if a valid date type
+                if (Object.prototype.toString.call(inputDate) === '[object Date]' && !isNaN(inputDate)){
+                    privateData.storedDate = inputDate;
```

Comment:
This works but keep in mind that by assigning an object like this, you're really assigning a reference to an object that could potentially be modified outside of your object, resulting in your internally stored Date instance getting modified at the same time. Better to just create your own new instance: privateData.date = new Date(inputDate);